test/parallel/test-debugger-util-regression.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+const path = require('path');
+const spawn = require('child_process').spawn;
+const assert = require('assert');
+
+const common = require('../common');
+
+const fixture = path.join(
+  common.fixturesDir,
+  'debugger-util-regression-fixture.js'
+);
+
+const args = [
+  'debug',
+  fixture
+];
+
+const proc = spawn(process.execPath, args, { stdio: 'pipe' });
+
+function cleanup() {
+  process.exitCode = 2;
+  proc.kill();
+}
+
+const timer = setTimeout(cleanup, common.platformTimeout(2000));
+
+let stdout = '';
+let stderr = '';
+
+let nextCount = 0;
+
+proc.stdout.on('data', (data) => {
+  stdout += data;
+  if (stdout.includes('> 1') && nextCount < 1 ||
+      stdout.includes('> 2') && nextCount < 2 ||
+      stdout.includes('> 3') && nextCount < 3) {
+    nextCount++;
+    proc.stdin.write('n\n');
+  }
+  else if (stdout.includes('{ a: \'b\' }')) {
+    clearTimeout(timer);
+    proc.stdin.write('.exit\n');
+  }
+  else if (stdout.includes('program terminated')) {
+    // Catch edge case present in v4.x
```

Comment:
Just curious, why don't you simply throw an exception here?  Moving the logic to the `process.on('exit')` listener means the stack trace will be less useful, it's not going to point to the actual offending code.